plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py

```diff
@@ -352,6 +361,40 @@ def register(
             except Exception as exc:  # pylint: disable=broad-except
                 self.__handle_exception(exc, idx, errorCallback)
 
+    def __validate_publish_entity_traits(self, entity_info, trait_ids, idx, error_callback):
+        if not bal.exists(entity_info, self.__library):
+            # Can publish with any traits if the entity doesn't exist.
+            return True
```

Comment:
[Question] Is this encoding specific behaviour into the interface? I wonder if the correct thing to do here would be to check what traits calling managementPolicy for write with a blank, or defaultEntityReference?
Maybe not, i'm not sure if managers are liable to want to depend on that sort of behaviour?

Reply:
It's the same behaviour that was there previously. But we can change it.

`defaultEntityReference` is alluring, though BAL doesn't currently support it - #53.

`managementPolicy` would work, but would in practice require users to add a `managementPolicy` `"exceptions"` entry for every trait set they plan to publish.  It's a bit of a misuse though, since we doc that hosts _can_ publish traits that aren't returned by `managementPolicy`  for `kWrite`, but the host can't expect those traits to be persisted.

:thinking: as an aside, we currently have no validation against `kRequired` policy, but that depends on trait introspection and documentation (i.e. which properties are required), so not really possible to implement in generic BAL.

I'd lean toward keeping the current behaviour as the default for when an entity isn't defined in the JSON at all. It can always be made explicit by adding an entry in the JSON (and in tests, you should always know the exact reference that will be used, even for non-existent entities).  I.e. unrestricted trait sets for non-existent entities seems like the most natural default case?

Reply:
Re our chat: leaving this as it is, although there is something to be said about how this gives BAL an opinion, but I'm deferring to your view that this is the only sensible opinion to have, at least in a test context.